Nordic: support for nRF5340 SoC in Zephyr

ioannisg opened the pull request:
nRF5340 basic support
- DTS, SoC, Board definitions
- BLE controller support

ioannisg committed:
dts: arm: nordic: nRF5340 Application MCU DTS structure

- Adding nRF5340 Application (CPU APP) common (Secure/Non-Secure)
peripheral DTS representation.
- Adding nrf5340 Application (CPU APP) default (Secure)
 .dtsi headers
- Adding nrf5340 Application (CPU APP) Non-Secure
 .dtsi headers
- Added nRF53 flash controller binding

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
soc: arm: nrf53: add nRF5340 CPUAPP SoC definition

Nordic nRF5340 CPUAPP (Application MCU) SoC definition
(Secure and Non-Secure domains).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

gamnes committed:
dts: arm: nordic: nRF5340 Network MCU DTS structure

Adding nRF5340 Network MCU (CPU NET) .dtsi headers

Signed-off-by: Gaute Gamnes <gaute.gamnes@nordicsemi.no>

ioannisg committed:
boards: arm: nordic: nRF5340_dk_nrf5340_cpuapp board definition

- Board cmake file with runner commands for Application MCU
- nRF5340 DK NRF5340 CPUAPP board definition (Application MCU)
  Secure and Non-Secure board definition
- nRF5340 shared SRAM partitioning info

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

gamnes committed:
soc: arm: nrf53: add nRF5340 CPUNET SoC definition

Nordic nRF5340 Network (CPU NET) SoC definition.

Signed-off-by: Gaute Gamnes <gaute.gamnes@nordicsemi.no>
Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

gamnes committed:
boards: arm: nordic: nRF5340_dk_nrf5340_cpunet board definition

- Modify board cmake file for Network MCU
- nRF5340 DK NRF5340 CPU NET board definition (Network MCU)

Signed-off-by: Gaute Gamnes <gaute.gamnes@nordicsemi.no>

ioannisg committed:
boards: nrf5340_dk_nrf5340: enable BT CTLR if building with BT

When building with support for BLE stack, enable
the BLE Controller module for the nRF5340 DK NRF5340
CPUNET (Network MCU).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: controller: include nRF5340-specific radio header

Conditionally include nrf5340_radio.h when building the
BLE controller for nRF5340 SoC. Introduce the header for
nrf5340.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
include: bluetooth: define hci macro for nRF53

Define the nRF53 HW variant in include/bluetooth/hci_vs.h
and pass the define in hci_vendor.h

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: controller: conditionally define NRF_RADIO_TXPOWER_POS4DBM

nRF5340 does not support a Radio TX power of 4dBm, so we
introduce a dependency for the respective Kconfig option
for TX power, so the option is never defined for nRF5340.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
drivers: entropy: set default RNG priority for nRF5340

For nRF5340, set the range and the default IRQ level
for the RNG interrupt.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: controller: enable support for nRF53 when building radio.c

Add the require #ifdef blocks in radio.c, in order to support
building for nRF53 SoC series (and, in particular, for nRF5340
CPU1 SoC).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: controller: setup DDPI configuration for nRF5340

This commit provides the DPPI configuration for the Radio
driver for nRF5340 SoC.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: controller: define SW IRQ structure for nRF53 SoCs

Extend the nRF5-specific SW IRQ framework to support nRF53.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: controller: enable support for nRF53 in Kconfig

This commit enables support in bluetooth/controller/Kconfig
for nRF53 series of SoCs.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: host: add nRF53 in HW variants

This commit adds a string representation of the
nRF53 variant in bluetooth host.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
bluetooth: controller: setup nRF53 DDPI configuration for Coded PHY

This commit provides the DPPI configuration for the Radio
driver for nRF5340 SoC, supporting LE Coded PHY.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

bfrog commented:
It's fun when you predict something and it comes true, but I'm very excited to get my hand on one of these!

mbolivar commented:
I can't build hello_world for either cpu0:

```
$ west build -b nrf5340_pca10095_cpu0
/home/mbolivar/zp/zephyr/boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu0.dts:8:10: fatal error: nordic/nrf5340_cpu0_qkaa.dtsi: No such file or directory
 #include <nordic/nrf5340_cpu0_qkaa.dtsi>
```

or cpu1:

```
$ west build -b nrf5340_pca10095_cpu1
devicetree error: unknown 'status' value "akay" in /soc/clock@41005000 in nrf5340_pca10095_cpu1.dts.pre.tmp, expected one of disabled, fail, okay, reserved, fail-sss (see the devicetree specification)
```

(I pulled this PR and ran west update before trying)

SebastianBoe commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu0_defconfig:
```diff
@@ -0,0 +1,24 @@
+CONFIG_BOARD_NRF5340_PCA10095_CPU0=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
```
Is it a given that the user wants to enable the MPU?

Does this automatically protect certain areas?

I've seen issues where one image accidentally protected some area that the next image used.

ioannisg replied:
@SebastianBoe this is standard configuration, i.e. building ARM platforms with MPU support if MPU is present.

> Does this automatically protect certain areas?
When enabling MPU at build time, SRAM is configured as PRIV RW, and Code as RO. This is basic generic configuration i.e. regardless of User Mode, or anything else.

>I've seen issues where one image accidentally protected some area that the next image used.

I agree that we might need to do something about it. E.g. at boot, we could consider clening up all MPU programming left-out from previous image. There is a similar issue for Stack Pointer clean-up. I 'll see if i can enhance that issue to cover MPU clean-up.

ioannisg replied:
@SebastianBoe https://github.com/zephyrproject-rtos/zephyr/issues/7404

ulfalizer commented on drivers/watchdog/Kconfig.nrfx:
```diff
@@ -12,3 +12,17 @@ config WDT_NRFX
+	bool "Enable WDT0 instance"
+	depends on HAS_HW_NRF_WDT || HAS_HW_NRF_WDT0
+	default y
+	 help
```
Accidental extra space.

Maybe the help text could spell out "Watchdog timer" too.

ulfalizer replied:
Compare with the `default` above.

ioannisg replied:
will be part of another PR

ulfalizer commented on drivers/watchdog/Kconfig.nrfx:
```diff
@@ -12,3 +12,17 @@ config WDT_NRFX
+	bool "Enable WDT1 instance"
+	depends on HAS_HW_NRF_WDT1
+	default n
+	 help
```
Accidental extra space.

ulfalizer replied:
Compare with the `default` above it.

ioannisg replied:
will be part of another PR

ulfalizer commented on dts/arm/nordic/nrf5340_cpu0_common.dtsi:
```diff
@@ -0,0 +1,328 @@
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	label="NRF_FLASH_DRV_NAME";
```
Missing spaces around `=`.

ulfalizer commented on soc/arm/nordic_nrf/nrf53/CMakeLists.txt:
```diff
@@ -0,0 +1,7 @@
+
+zephyr_sources(
+  soc.c
+  )
```
Maybe `zephyr_sources(soc.c)` to shave some lines (though I get the idea).

ulfalizer reviewed (commented):
Some random unimportant nits.

ioannisg commented on soc/arm/nordic_nrf/nrf53/Kconfig.series:
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
```
It's exactly the same with nRF51,52,and 91 headers in the respective files, so I'd keep that as is  for now.

ioannisg replied:
We can clean up in an exclusive clean-up PR:; i'll make a note of it.

ioannisg commented on soc/arm/nordic_nrf/nrf53/Kconfig.series:
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config  SOC_SERIES_NRF53X
```
Fixed that, thanks.

ioannisg commented on soc/arm/nordic_nrf/nrf53/Kconfig.soc:
```diff
@@ -0,0 +1,104 @@
+	select HAS_HW_NRF_WDT
+
+choice
+prompt "nRF53x MCU Selection"
```
Thanks, fixed that as well.

ioannisg reviewed (commented):
Addressing feedback by @ulfalizer in SoC, Board, DTS changes.

ioannisg commented:
> 0: fatal error: nordic/nrf5340_cpu0_qkaa.dtsi: No such file or directory
>  #include <nordic/nrf5340_cpu0_qkaa.dtsi>

@mbolivar you should be able to build now - I had forgotten to rename the DTS headers.

anangl commented on dts/arm/nordic/nrf5340_cpu0_common.dtsi:
```diff
@@ -0,0 +1,328 @@
+	 * compatible = "nordic,nrf-twim" or
+	 *              "nordic,nrf-twis".
+	 */
+	compatible = "nordic,nrf-i2c";
```
```suggestion
```
This property is to be chosen at some further level, in a board definition, for instance.

anangl commented on dts/arm/nordic/nrf5340_cpu0_common.dtsi:
```diff
@@ -0,0 +1,328 @@
+	 * compatible = "nordic,nrf-twim" or
+	 *              "nordic,nrf-twis".
+	 */
+	compatible = "nordic,nrf-i2c";
```
```suggestion
```

ioannisg replied:
done

anangl commented on dts/arm/nordic/nrf5340_cpu0_common.dtsi:
```diff
@@ -0,0 +1,328 @@
+	 * compatible = "nordic,nrf-spim" or
+	 *              "nordic,nrf-spis".
+	 */
+	compatible = "nordic,nrf-spi";
```
```suggestion
```

anangl commented on dts/arm/nordic/nrf5340_cpu0_common.dtsi:
```diff
@@ -0,0 +1,328 @@
+	 * compatible = "nordic,nrf-spim" or
+	 *              "nordic,nrf-spis".
+	 */
+	compatible = "nordic,nrf-spi";
```
```suggestion
```

anangl commented on dts/arm/nordic/nrf5340_cpu0_qkaa.dtsi:
```diff
@@ -0,0 +1,22 @@
+
+/ {
+	soc {
+		compatible = "nordic,nRF5340-CPU0-QIAA", "nordic,nRF5340-CPU0", "nordic,nRF53", "simple-bus";
```
`-QKAA`

ioannisg replied:
Fixing

anangl commented on dts/arm/nordic/nrf5340_cpu0ns_qkaa.dtsi:
```diff
@@ -0,0 +1,22 @@
+
+/ {
+	soc {
+		compatible = "nordic,nRF5340-CPU0-QIAA", "nordic,nRF5340-CPU0", "nordic,nRF53", "simple-bus";
```
`-QKAA`

ioannisg replied:
Fixing

anangl commented on dts/arm/nordic/nrf5340_cpu1_qkaa.dtsi:
```diff
@@ -0,0 +1,22 @@
+
+/ {
+	soc {
+		compatible = "nordic,nRF5340-CPU1-QIAA", "nordic,nRF5340-CPU1", "nordic,nRF53", "simple-bus";
```
`-QKAA`

ioannisg replied:
Fixing, thanks.

anangl commented on boards/arm/nrf5340_pca10095/Kconfig.board:
```diff
@@ -0,0 +1,23 @@
+endif # SOC_NRF5340_CPU0_QKAA
+
+
+if SOC_NRF5340_CPU1_QIAA
```
`_QKAA`

ioannisg replied:
Already fixed

mbolivar commented:
> @mbolivar you should be able to build now - I had forgotten to rename the DTS headers.

Yep, just down to a DT warning for cpu0, but both builds are working for me.

```
nrf5340_pca10095_cpu1.dts.pre.tmp:77.25-83.5: Warning (unique_unit_address_if_enabled): /soc/clock@41005000: duplicate unit-address (also used in node /soc/power@41005000)
```

ioannisg commented on dts/arm/nordic/nrf5340_cpu1.dtsi:
```diff
@@ -0,0 +1,251 @@
+		};
+
+		i2c0: i2c@41013000 {
+			compatible = "nordic,nrf-i2c";
```
I'll apply the suggestion.

ioannisg commented on dts/arm/nordic/nrf5340_cpu1.dtsi:
```diff
@@ -0,0 +1,251 @@
+		};
+
+		spi0: spi@41013000 {
+			compatible = "nordic,nrf-spi";
```
alright

ioannisg reviewed (commented):
Applied the feedback from @anangl -

joerchan commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -164,6 +164,7 @@ config BT_CTLR_TX_PWR_PLUS_5
 
 config BT_CTLR_TX_PWR_PLUS_4
 	bool "+4 dBm"
+	depends on SOC_SERIES_NRF51X || SOC_COMPATIBLE_NRF52X
```
Can we make something like TX_PWR_PLUS_4_SUPPORT, and have nrf521x and nrf52x select them instead?

kl-cruz replied:
checks for each CPU is hard to maintain

ioannisg replied:
Yes, I also agree on that.

ioannisg replied:
Mind however, @joerchan @kl-cruz that this needs to be done for all power levels that are SoC specific. Could we avoid to do this in the current PR? IF so, I can open an issue to introduce thes "capabilities" to the nRF SoCs

joerchan replied:
> Could we avoid to do this in the current PR?

Yes, absolutely

kl-cruz replied:
Sure

joerchan commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -306,7 +307,7 @@ config BT_CTLR_DATA_LENGTH
 	bool
 	depends on BT_DATA_LEN_UPDATE
-	default y if SOC_COMPATIBLE_NRF52X || BT_CTLR_DATA_LENGTH_CLEAR
+	default y if (SOC_COMPATIBLE_NRF52X || SOC_SERIES_NRF53X) || BT_CTLR_DATA_LENGTH_CLEAR
```
Heads up about conflicting changes here:  https://github.com/zephyrproject-rtos/zephyr/pull/19710#discussion_r334458656

ioannisg replied:
Thanks. For now I think we can live with this, but I agree we should try to have a "capability" Kconfig symbol for Data Length Extension 
I'll make a note to file an issue for that.

joerchan replied:
@ioannisg They are already doing that change. So we just have to use it once it is merged. It should make this a lot easier.

joerchan commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:
```diff
@@ -20,9 +20,21 @@
 #if defined(CONFIG_SOC_SERIES_NRF51X)
 #define RADIO_PDU_LEN_MAX (BIT(5) - 1)
-#elif defined(CONFIG_SOC_COMPATIBLE_NRF52X)
+#elif defined(CONFIG_SOC_COMPATIBLE_NRF52X) || \
```
make it `#else`?

ioannisg replied:
Related to your above comment, @joerchan. We should probably have an ifdef here coupled with the "capability" for DLE.

But for now, I would prefer to leave as is, cause this is more **educative**, about which platforms really can support DLE.

ioannisg replied:
Yes, I agreed. We should send a refactoring PR with DLE feature support. But not as part of this PR.

joerchan replied:
@ioannisg I would agree with you, but this is not about supporting DLE. This is about supporting an older chip that was made when the length field of the LL PDU was only 5 bits. The length field is gonna be 8 bits whether or not you support DLE or not. Which is why I think my suggestion make sense, and is easier to maintain. (Same for CCM Extended length).

SebastianBoe replied:
> We should send a refactoring PR with DLE feature support. But not as part of this PR.

Refactoring the existing codebase to be able to support a new feature should be a part the feature-PR.

ioannisg replied:
Right, I see what you mean - you want the length of the field to be tight to the SoC and not the larger PDU length extension. Then, I 'll send a PR upstream and remove the change from here.

ioannisg replied:
https://github.com/zephyrproject-rtos/zephyr/pull/20304
If the above goes through, then this will be automatically resolved. Otherwise, we keep current solution.

joerchan commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:
```diff
@@ -962,7 +978,7 @@ void *radio_ccm_rx_pkt_set(struct ccm *ccm, u8_t phy, void *pkt)
 	       CCM_MODE_MODE_Msk;
 
-#if defined(CONFIG_SOC_COMPATIBLE_NRF52X)
+#if defined(CONFIG_SOC_COMPATIBLE_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
```
Just make it `!defined(NRF51x)`?

ioannisg replied:
Same argument as above @joerchan, making it !NRF51 makes this code default for any other SoC and also, not educative when reading the code. When reading the code, IMHO, it is better to be able to see which platforms can enable the compilation of a code block (not the reverse).

No strong opinion though. I can change that if you insist.

ioannisg replied:
Will be automatically resolved after merging #20304

ioannisg committed:
scripts: west_commands: nrfjprog: extend nrfjprog.py runner for nrf53

This commit extends the nrfjprog.py runner script so it adds
support for nRF53. In the wake of the changes done in the
runner script, we extend the testing done in test_nrfjprog.py,
adding the required coverage for the nRF53.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
@mbolivar can you review the changes in the nrfjprog runner, the board.cmake and the related test extension?

anangl commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu0_common.dts:
```diff
@@ -0,0 +1,171 @@
+};
+
+&uart1 {
+	status = "okay";
```
Instead of `uart1`, you should enable `i2c1` (with `compatible = "nordic,nrf-twim";`), otherwise nothing the uses `I2C_1` (which is enabled in `boards/arm/nrf5340_pca10095/Kconfig.defconfig`) can be built successfully.

ioannisg replied:
agreed, doing it now

anangl commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu0_common.dts:
```diff
@@ -0,0 +1,171 @@
+	rts-pin = <36>;
+	cts-pin = <37>;
+};
+
```
You should also enable `spi2` here, with `compatible = "nordic,nrf-spim";`, as `boards/arm/nrf5340_pca10095/Kconfig.defconfig` sets `SPI_2` to `y`.

ioannisg replied:
Right, done that as well.

anangl commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu1.yaml:
```diff
@@ -0,0 +1,13 @@
+ram: 64
+flash: 256
+supported:
+  - i2c
```
No I2C instance is enabled for this board, so this entry should be removed.

ioannisg replied:
removing, thanks.

mbolivar commented on scripts/west_commands/runners/nrfjprog.py:
```diff
@@ -103,8 +113,13 @@ def do_run(self, command, **kwargs):
+                '--snr', board_snr]
+        else:
+            program_cmd = ['nrfjprog', '--program', self.hex_, '-f', self.family,
+                '--snr', board_snr] + self.tool_opt
```
tool_opt should be available for all families, not just NRF53.

ioannisg replied:
OK, I am making this available for all platforms

mbolivar commented on scripts/west_commands/runners/nrfjprog.py:
```diff
@@ -14,14 +15,19 @@
 
-    def __init__(self, cfg, family, softreset, snr, erase=False):
+    def __init__(self, cfg, family, softreset, snr, erase=False,
+            tool_opt=[]):
```
```suggestion
                 tool_opt=[]):
```

ioannisg replied:
fixing

mbolivar commented on scripts/west_commands/runners/nrfjprog.py:
```diff
@@ -42,11 +48,15 @@ def do_add_parser(cls, parser):
                             help='serial number of board to use')
+        parser.add_argument('--tool-opt', default=[], action='append',
+                            help='''Additional options for nrfjprog,
+                    e.g. \'-coprocessor\' ''')
```
```suggestion
                            e.g. "--recover"''')
```

mbolivar replied:
This suggestion combines a whitespace nit with a suggested change that applies to more families.

ioannisg replied:
Sure, I'll fix that.

mbolivar reviewed (changes requested):
Just some nits.

Please squash the commit fixing test_nrfjprog.py into the one that changed nrfjprog.py.

anangl commented on boards/arm/nrf5340_pca10095/Kconfig:
```diff
@@ -0,0 +1,18 @@
+# BT_CTLR depends on BT. When BT is enabled we should default to also
+# enabling the controller.
+config BT_CTLR
+	default y if BT
```
Why are you actually doing it here, and not in `Kconfig.defconfig`, like it is done for other boards (well, with the exception of `nrf52810_pca10040`, that's perhaps why)?

ioannisg replied:
Yes, I didn't see a consistency here, so felt free to add it here.

anangl replied:
I won't insist on doing it now. It can be corrected later, together with the same correction for `nrf52810_pca10040`.

anangl commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu0_common.dts:
```diff
@@ -0,0 +1,177 @@
+	compatible = "nordic,nrf-twim";
+	status = "okay";
+	sda-pin = <30>;
+	scl-pin = <31>;
```
Are these just some place-holder pins? I'm asking because same ones are used for `led2` and `led3`.

ioannisg replied:
Yes, that's OK for now, I believe. Unless there is a reason to change them.

anangl commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu1.dts:
```diff
@@ -0,0 +1,149 @@
+};
+
+&ipc {
+	status = "okay";
```
This is redundant, this property is already set to "okay" in [dts/arm/nordic/nrf5340_cpu1.dtsi](https://github.com/zephyrproject-rtos/zephyr/pull/20157/files#diff-031e9ff221884ea236023201f93a779eR144).

ioannisg replied:
Right. Will remove. (no harm, though)

ioannisg replied:
Will remove from cpu0 as well.

ioannisg replied:
(if it exists)

anangl commented on soc/arm/nordic_nrf/nrf53/soc.c:
```diff
@@ -0,0 +1,79 @@
+#include <kernel.h>
+#include <init.h>
+#include <cortex_m/exc.h>
+#include <nrfx.h>
```
This is redundant. `<soc.h>` includes `<nrfx.h>`.

anangl commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h:
```diff
@@ -48,4 +48,45 @@ static inline void hal_swi_job_pend(void)
+
+/* Split architecture uses max. two SWI */
+#define HAL_SWI_RADIO_IRQ     NRF5_IRQ_EGU0_IRQn
+#define HAL_SWI_WORKER_IRQ    NRF5_IRQ_RTC0_IRQn
```
Definitions of these two `NRF5_IRQ_*` labels were removed by [this commit](https://github.com/zephyrproject-rtos/zephyr/commit/35754375bc2447b4cd4c429d2662a2e96ed5b8cd).
Use `EGU0_IRQn` and `RTC0_IRQn` instead.

ioannisg replied:
Great, fixing this as well.

anangl commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h:
```diff
@@ -48,4 +48,45 @@ static inline void hal_swi_job_pend(void)
+#define HAL_SWI_WORKER_IRQ    NRF5_IRQ_RTC0_IRQn
+
+#if !defined(CONFIG_BT_CTLR_LOW_LAT) && \
+	defined(CONFIG_BT_CTLR_ULL_HIGH_PRIO == CONFIG_BT_CTLR_ULL_LOW_PRIO)
```
This use of `defined` seems to be incorrect.

ioannisg replied:
Fixing, thanks.

ioannisg commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu0_common.dts:
```diff
@@ -0,0 +1,177 @@
+	ch0-pin = <28>;
+};
+
+&spi3 {
```
Right. That's why CI is failing. I caught it as well.

ioannisg commented on boards/arm/nrf5340_pca10095/nrf5340_pca10095_cpu0_common.dts:
```diff
@@ -0,0 +1,177 @@
+};
+
+&spi3 {
+	compatible = "nordic,nrf-spim";
```
Right. WIll remove.

mbolivar commented:
Code changes look good on the runner side. Please do the squash I mentioned earlier though.

ioannisg commented:
> > Code changes look good on the runner side. Please do the squash I mentioned earlier though.
> 
> Right. I want a separate commit for nrfjprog script changes. So I'll take this out of the _cpu0 board definition and put all script changes into 1 commit

Done now @mbolivar 
All changes related to the runner script and the testing of it are lying in a single commit.

mbolivar commented:
> Done now @mbolivar
> All changes related to the runner script and the testing of it are lying in a single commit.

OK, but please move that commit to before the board definition commits which make use of it.

ioannisg commented:
> > Done now @mbolivar
> > All changes related to the runner script and the testing of it are lying in a single commit.
> 
> OK, but please move that commit to before the board definition commits which make use of it.

Done now @mbolivar :)

mbolivar reviewed (approved):
Runner changes look good to me.

The west.yml changes are OK for a draft but of course need an update once https://github.com/zephyrproject-rtos/hal_nordic/pull/11 can go in.

cvinayak reviewed (dismissed):
I am not able to build the samples/bluetooth/peripheral for nrf5340_pca10095_cpu1 board. I am getting NRF_AAR and NRF_ECB as undeclared.

kl-cruz commented on boards/arm/nrf52840_pca10059/board.c:
```diff
@@ -16,7 +16,8 @@ static int board_nrf52840_pca10059_init(struct device *dev)
 	 * Increase GPIO voltage to 3.0 volts.
 	 */
-	if ((nrf_power_mainregstatus_get() == NRF_POWER_MAINREGSTATUS_HIGH) &&
+	if ((nrf_power_mainregstatus_get(NRF_POWER) ==
```
nrfx 2.0 introduces passing the pointer to the structure of registers of the peripheral in every HAL function

ioannisg committed:
boards: arm: nrf5340 dk: add documentation for nRF5340 DK board

We add supportive documentation for the nRF5340 DK board.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
@cvinayak can you re-visit this?

ioannisg commented:
@nordic-krch @barsok @anangl @cvinayak @carlescufi please review!

dbkinder commented on boards/arm/nrf5340_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,259 @@
+The nRF5340 DK NRF5340 is a single-board development kit for evaluation
+and development on the Nordic nRF5340 System-on-Chip (SoC).
+
+nRF5340 is a dual-core SoC based on the Arm® Cortex®-M33 architecture, with
```
Some formatting fixes:
```
The nRF5340 is a dual-core SoC based on the Arm® Cortex®-M33 architecture, with:

* a full-featured ARM Cortex-M33F core with DSP instructions, FPU, and
  ARMv8-M Security Extension, running at up to 128 MHz, referred to as
  the **Application MCU**
* a secondary ARM Cortex-M33 core, with a reduced feature set, running at
  a fixed 64 MHz, referred to as the **Network MCU**.
```

ioannisg replied:
@dbkinder thanks for the review, suggestion applied - please do a quick pass :)

dbkinder reviewed (changes requested):
LGTM with one fix

carlescufi commented on boards/arm/nrf5340_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,260 @@
+Overview
+********
+
+The nRF5340 DK NRF5340 is a single-board development kit for evaluation
```
Replace with nRF5340 DK, remove the extra NRF5340. The DK is called nRF5340 DK.

ioannisg replied:
Alright, will do.

carlescufi commented on boards/arm/nrf5340_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,260 @@
+
+* a full-featured ARM Cortex-M33F core with DSP instructions, FPU, and
+  ARMv8-M Security Extension, running at up to 128 MHz, referred to as
+  the **Application MCU**
```
MCU or CPU or Core?

ioannisg replied:
MCU

carlescufi commented on boards/arm/nrf5340_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,260 @@
+  ARMv8-M Security Extension, running at up to 128 MHz, referred to as
+  the **Application MCU**
+* a secondary ARM Cortex-M33 core, with a reduced feature set, running at
+  a fixed 64 MHz, referred to as the **Network MCU**.
```
MCU or CPU or Core?

ioannisg replied:
So and ARM Cortex-M is a core, obviously, but the "core" names are Application MCU and NEtwork MCU and are consistent through this documentation (or should be)

nordic-krch commented on boards/arm/nrf5340_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,260 @@
+Hardware
+********
+
+nRF5340 PDK has two external oscillators. The frequency of
```
PDK or DK?

ioannisg replied:
Should be DK - that's a copy paste error, thanks.

ioannisg replied:
Will fix

cvinayak reviewed (approved):
Tested, I see ADV_IND on my nRF Sniffer!

nordic-krch commented:
@ioannisg patch looks good, what about checkpatch complains? I see that some (checked few at the bottom) are valid and should be fixed.

ioannisg commented:
> @ioannisg patch looks good, what about checkpatch complains? I see that some (checked few at the bottom) are valid and should be fixed.

Yes, I can fix some of those :)

ioannisg commented:
> @ioannisg patch looks good, what about checkpatch complains? I see that some (checked few at the bottom) are valid and should be fixed.

@nordic-krch I've resolved (hopefully) all checkpatch warnings in radio_nrf_ppi.h and most of the warnings in soc.h, soc.c
Didn't resolve warnings in .dts files; seems the line exceeds , there, are for a reason :)

ioannisg commented:
@carlescufi @dbkinder pls, do one more review round of the ./doc . Thanks!

ioannisg reviewed (commented):
@ulfalizer your comments in #20374 are addressed in this PR; please, go through them here.

ulfalizer commented:
> @ulfalizer your comments in #20374 are addressed in this PR; please, go through them here.

Hmm... don't see the updated binding description and headers.

ulfalizer commented on boards/arm/nrf5340_dk_nrf5340/Kconfig:
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
```
```
# Kconfig - nRF5340 DK NRF5340 board configuration
# Copyright (c) 2019 Nordic Semiconductor ASA
(blank line here)
# SPDX-License-Identifier: Apache-2.0
```

(Nit from old PR.)

ioannisg replied:
Ah right, so you want a blank line completely, not with `#` :)

ioannisg replied:
OK @ulfalizer hold on :)

ulfalizer replied:
Skip the `Kconfig - ...` part too. It's clear already that it's about Kconfig.

Won't keep nitting these for all PRs. It's mostly to indirectly make people aware that there's some consistency now.

ioannisg replied:
@ulfalizer done now :)

ioannisg replied:
Oh, wait, a sec, this may be missing.

ulfalizer replied:
Note that there's also a bunch of other similar headers to be fixed (e.g. in boards/arm/nrf5340_dk_nrf5340/Kconfig.board).

ioannisg replied:
Fixed now, thanks :)

ulfalizer commented on dts/bindings/flash_controller/nordic,nrf53-flash-controller.yaml:
```diff
@@ -0,0 +1,8 @@
+title: Nordic NVMC
+
+description: >
+    This binding gives a base representation of the Nordic NVMC
```
Could give a more meaningful description. Not sure what "base representation" was ever supposed to mean.

Something like

```
description: |
    Nordic NVMC (<explain acronym>). See datasheet at ...
```

Or whatever other information might be useful.

`|` instead of `>` preserves newlines. The description shows up as a comment in the header above the generated macros.

ioannisg replied:
Fixed the `|` instead of `<`

Now, I really don't want to touch the text this in this PR because this binding is identical to the bindings of nrf51, nrf52, and nrf91.

ulfalizer commented:
> Now, I really don't want to touch the text this in this PR because this binding is identical to the bindings of nrf51, nrf52, and nrf91.

Please change it. The `description` text for most bindings is pointless now due to that string just getting copied around, so the consistency needs to be broken.

ioannisg commented:
> > Now, I really don't want to touch the text this in this PR because this binding is identical to the bindings of nrf51, nrf52, and nrf91.
> 
> Please change it. The `description` text for most bindings is pointless now due to that string just getting copied around, so the consistency needs to be broken.

No, but I will ask @nvlsianpu or @anangl to send a PR that change all 4 nrf flash-controller bindings.
(I agree completely with your request, but there should be another workflow )

ulfalizer reviewed (commented):
> No, but I will ask @nvlsianpu or @anangl to send a PR that change all 4 nrf flash-controller bindings.
> (I agree completely with your request, but there should be another workflow )

Fine with me, as long as it gets done. Won't hurt to change it here to start with though. No description strings like that should ever be added from now on if we can avoid it I think.

ulfalizer reviewed (approved):
Nits fixed.

ioannisg commented:
> > No, but I will ask @nvlsianpu or @anangl to send a PR that change all 4 nrf flash-controller bindings.
> > (I agree completely with your request, but there should be another workflow )
> 
> Fine with me, as long as it gets done. Won't hurt to change it here to start with though. No description strings like that should ever be added from now on if we can avoid it I think.

#20412 FYI

kapi-no reviewed (approved):
one typo

carlescufi reviewed (approved):
LGTM now, thank you

carlescufi commented:
@ioannisg please rebase

galak reviewed (dismissed):
Marking request changes because of what looks like dangerous changes in `scripts/ci/run_ci.sh`

ioannisg commented:
@dbkinder ping for minor revision

nvlsianpu commented on boards/arm/nrf5340_dk_nrf5340/nrf5340_dk_nrf5340_cpunet.dts:
```diff
@@ -0,0 +1,145 @@
+
+		boot_partition: partition@1000000 {
+			label = "mcuboot";
+			reg = <0x01000000 0xc000>;
```
please use relative partition offset: 0x0 here, 0xc000 below and so on

nvlsianpu replied:
After discussion off-channel decoded to keep as is

galak commented on boards/arm/nrf5340_dk_nrf5340/nrf5340_dk_nrf5340_cpuapp.yaml:
```diff
@@ -0,0 +1,14 @@
+flash: 256
+supported:
+  - i2c
+  - pwm
```
Seems like a small list of supported features?

ioannisg replied:
For now yes, there will be more coming up, as we complete the support for this SoC

galak commented on soc/arm/nordic_nrf/nrf53/dts_fixup.h:
```diff
@@ -0,0 +1,42 @@
+#define DT_SPI_1_NAME			DT_NORDIC_NRF_SPI_SPI_1_LABEL
+#define DT_SPI_2_NAME			DT_NORDIC_NRF_SPI_SPI_2_LABEL
+
+#define DT_WDT_0_NAME			DT_NORDIC_NRF_WATCHDOG_WDT_0_LABEL
```
Nothing uses this define `DT_WDT_0_NAME`, can we remove it?

galak reviewed (commented):
Looks good to me, one question on the DT_WDT_0_NAME and if we can remove it.  Would prefer the Bluetooth changes in a different PR, but the SoC/Board stuff is good.

dbkinder commented on boards/arm/nrf5340_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,260 @@
+* a secondary ARM Cortex-M33 core, with a reduced feature set, running at
+  a fixed 64 MHz, referred to as the **Network MCU**.
+
+The nrf5340_dk_nrf5340_cpuapp provides support for the Application MCU on
```
Given @carlescufi question about the Application and Network **MCU** name, it's amusing that the board name has **cpu** in it (instead of **mcu**). Oh well.

carlescufi replied:
Yes, I dislike that but it's for consistency with other existing multi-core boards.

dbkinder reviewed (approved):
Given @carlescufi question about the Application and Network **MCU** name, it's amusing that the board name has **cpu** in it (instead of **mcu**). Oh well.  Otherwise, docs LGTM.
